tests/integ/sagemaker/workflow/test_workflow.py

```diff
@@ -1167,3 +1169,65 @@ def test_load_method(sagemaker_session, role, pipeline_name, region_name):
             pipeline.delete()
         except Exception:
             pass
+
+
+def test_get_last_execution(sagemaker_session, role, pipeline_name, region_name):
+    param = ParameterInteger(name="MyInt")
+    cond = ConditionEquals(left=param, right=1)
+    step_fail = FailStep(
+        name="FailStep",
+        error_message="Failed due to hitting in else branch",
+    )
+    step_cond = ConditionStep(
+        name="CondStep",
+        conditions=[cond],
+        if_steps=[],
+        else_steps=[step_fail],
+    )
+    pipeline = Pipeline(
+        name=pipeline_name,
+        steps=[step_cond],
+        sagemaker_session=sagemaker_session,
+        parameters=[param],
+    )
+
+    try:
+        response = pipeline.create(role)
+        create_arn = response["PipelineArn"]
+        assert re.match(
+            rf"arn:aws:sagemaker:{region_name}:\d{{12}}:pipeline/{pipeline_name}",
+            create_arn,
+        )
+        assert len(json.loads(pipeline.describe()["PipelineDefinition"])["Steps"]) == 1
+
+        execution_success = pipeline.start(
+            parameters={
+                "MyInt": 1,
+            }
+        )
+        execution_success.wait()
+
+        execution_fail = pipeline.start(
+            parameters={
+                "MyInt": 3,
+            }
+        )
+
+        time.sleep(10)
```

Comment:
Please follow this example: https://github.com/aws/sagemaker-python-sdk/blob/master/tests/integ/sagemaker/workflow/test_fail_steps.py#L260-L263
to make sure the execution completely finishes
We can add the try-catch blocks for each of the execution